nvi-commons/src/main/java/no/sikt/nva/nvi/common/service/NviService.java

```diff
@@ -45,134 +35,113 @@ public static NviService defaultNviService() {
+    private static List<DbApprovalStatus> generatePendingApprovalStatuses(List<DbInstitutionPoints> institutionUris) {
         return institutionUris.stream()
-                   .map(uri -> ApprovalStatus.builder()
-                                   .withStatus(Status.PENDING)
-                                   .withInstitutionId(uri)
-                                   .build())
+                   .map(institutionPoints -> DbApprovalStatus.builder()
+                                                 .status(DbStatus.PENDING)
+                                                 .institutionId(institutionPoints.institutionId())
+                                                 .build())
                    .toList();
     }
 
-    private static Candidate toPendingCandidate(CandidateEvaluatedMessage candidateEvaluatedMessage,
-                                                Map<URI, BigDecimal> institutionPoints) {
-        return Candidate.builder()
-                   .withPublicationBucketUri(candidateEvaluatedMessage.publicationBucketUri())
-                   .withPublicationId(candidateEvaluatedMessage.candidateDetails().publicationId())
-                   .withIsApplicable(true)
-                   .withCreators(mapToVerifiedCreators(candidateEvaluatedMessage.candidateDetails().verifiedCreators()))
-                   .withLevel(Level.parse(candidateEvaluatedMessage.candidateDetails().level()))
-                   .withInstanceType(candidateEvaluatedMessage.candidateDetails().instanceType())
-                   .withPublicationDate(mapToPublicationDate(candidateEvaluatedMessage.candidateDetails()
-                                                                 .publicationDate()))
-                   .withPoints(mapToInstitutionPoints(institutionPoints))
-                   .withApprovalStatuses(generatePendingApprovalStatuses(institutionPoints.keySet()))
-                   .build();
+    @JacocoGenerated // bug in jacoco report that is unable to exhaust the switch. Should be fixed in version 0.8.11
+    private DbApprovalStatus toUpdatedApprovalStatus(DbApprovalStatus oldApprovalStatus,
+                                                     DbApprovalStatus newApprovalStatus) {
+        return switch (newApprovalStatus.status()) {
+            case APPROVED, REJECTED -> oldApprovalStatus.copy()
+                                           .status(newApprovalStatus.status())
+                                           .finalizedBy(newApprovalStatus.finalizedBy())
+                                           .finalizedDate(Instant.now())
+                                           .build();
+            case PENDING -> oldApprovalStatus.copy()
+                                .status(DbStatus.PENDING)
+                                .finalizedBy(null)
+                                .finalizedDate(null)
+                                .build();
+        };
```

Comment:
Maybe refactor to:
-> updateStatus(oldApprovalStatus, newApprovalStatus)
->  newPendingStatus(newApprovalStatus)